src/htm/algorithms/Connections.cpp

```diff
@@ -122,7 +141,17 @@ Synapse Connections::createSynapse(Segment segment,
   for (const Synapse& syn : synapsesForSegment(segment)) {
     const CellIdx existingPresynapticCell = dataForSynapse(syn).presynapticCell; //TODO 1; add way to get all presynaptic cells for segment (fast)
     if (presynapticCell == existingPresynapticCell) {
-      return syn; //synapse (connecting to this presyn cell) already exists on the segment; don't create a new one, exit early and return the existing
+      //synapse (connecting to this presyn cell) already exists on the segment; don't create a new one, exit early and return the existing
+      NTA_ASSERT(synapseExists_(syn));
+      //TODO what is the strategy on creating a new synapse, while the same already exists (on the same segment, presynapticCell) ??
+      //1. just keep the older (former default)
+      //2. throw an error (ideally, user should not createSynapse() but rather updateSynapsePermanence())
+      //3. create a duplicit new synapse -- NO. This is the only choice that is incorrect! HTM works on binary synapses, duplicates would break that.
+      //4. update to the max of the permanences (default)
```

Comment:
documented and changed the default strategy for `createSynapse()` if such synapse already exists. All four alternatives discussed above, 
- former default: leave as is; 
- now: update permanence if the new perm would be larger.

Reply:
...impact is not really seen in our unit-tests as none trigger this "duplicit synapse created" branch.